ide/editor.util/src/org/netbeans/lib/editor/util/swing/DocumentUtilities.java

```diff
@@ -65,9 +65,9 @@
     
     private static final Object TYPING_MODIFICATION_KEY = new Object();
     
-    private static Field numReadersField;
+    private static volatile Field numReadersField;
```

Comment:
BTW is this necessary, strictly speaking ? The `nuimReadersField` may be eventually initialized several times (a thread that was already started at the time of the previous assignment could see `null`) - but the Field instance should point to the same data (?).  But bcs of number of invocations (and other heavy locking on Document) I guess it makes no difference between volatile/nonvolatile.

Reply:
I agree that the synchronization is wrong. Looking at it, I'll adjust it a bit...

Reply:
I've removed the wrong synchronization, it's up to the volatile flag only now. It can initialize multiple times, but after that it's just one volatile access.